constellation-app/constellation
# Save custom icons into the star file

### Prerequisites


    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change

Custom icons that have been added to graphs are now included in the data saved into the star files.
Loading the star file on another machine will now extract and display those custom icons.



### Alternate Designs
N/A



### Why Should This Be In Core?
Has been requested by some prominent users.



### Benefits

Graphs can be shared between users, with all images intact.



### Possible Drawbacks

The star files will be notably larger with the addition of custom icon images stored within.


### Verification Process

1. Create a graph with some custom images used on the nodes.
2. Save the graph
3. Copy the star file to a second machine running Constellation.
4. Load the file on the second machine.
5. Confirm the custom images are displayed in the graph.
6. Confirm that when assigning an icon to a node, all the custom icons are now available in the User-Defined section.

Also to be confirmed: If loading a graph with custom icons that match with existing custom icons already on the user's machine, the icons will only be overwritten if the star file contains the more recently updated versions of the icons.



### Applicable Issues

#1121
